hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -261,6 +268,44 @@ private[hudi] object HoodieSparkSqlWriter {
     hiveSyncConfig
   }
 
+  private def metaSync(parameters: Map[String, String],
+                       basePath: Path,
+                       hadoopConf: Configuration): Boolean = {
+    val hiveSyncEnabled = parameters.get(HIVE_SYNC_ENABLED_OPT_KEY).exists(r => r.toBoolean)
+    var metaSyncEnabled = parameters.get(HUDI_SYNC_ENABLED_OPT_KEY).exists(r => r.toBoolean)
+    var syncClientToolClass = parameters(SYNC_CLIENT_TOOL_CLASS)
+    // for backward compatibility
+    if (hiveSyncEnabled) {
+      metaSyncEnabled = true
+      syncClientToolClass = String.format("%s,%s", syncClientToolClass, "org.apache.hudi.hive.HiveSyncTool")
```

Comment:
if someone does `hiveSyncEnabled == true && metaSyncEnabled == true && syncClientToolClass = org.apache.hudi.hive.HiveSyncTool`, we will sync two times? can we just a set to hold the classes.

Reply:
ditto

Reply:
done